packages/medusa/src/services/fulfillment.ts

```diff
@@ -224,11 +234,34 @@ class FulfillmentService extends TransactionBaseService {
 
       const created = await Promise.all(
         fulfillments.map(async ({ shipping_method, items }) => {
+          await pvInventoryServiceTx.validateInventoryAtLocation(
+            items,
+            locationId!
+          )
+
+          await Promise.all(
+            items.map(async (i) => {
+              await pvInventoryServiceTx.adjustReservationsQuantityByLineItem(
+                i.id,
+                i.variant_id!,
+                locationId!,
+                -i.quantity
+              )
+
+              await pvInventoryServiceTx.adjustInventory(
+                i.variant_id!,
+                locationId!,
+                -i.quantity
+              )
```

Comment:
suggestion: guard these calls with an `if (i.variant_id)` - that will allow us to clean up the PVInventoryService's method signatures.